.gitlab-ci.yml

```diff
@@ -79,6 +82,39 @@ Setup:Review:
           $CI_COMMIT_REF_NAME !~  "/^nodeploy\/.*$/"
       when: always
 
+# Linting
+Lint:
+  stage: lint
+  before_script:
+    - pip install -r requirements.txt
+  script:
+    - flake8 .
+  only:
+    - branches
+    - merge_requests
+
+# Formatting
+Format:
```

Comment:
Without a push command this stage will correct the code in the runner but nothing else.

Personally I don't think it is a good idea to run code modification stages in CI. 
My reasons are :-
- The author ends up with a slightly different version of the code to the PR
- Can introduce hard to discover errors 

Black should be ran as a precommit hook or on save in your IDE.